src/rpp/sockets.cpp

```diff
@@ -1958,6 +1993,48 @@ namespace rpp
         return get_ip_interface(network_interface, af).broadcast.str();
     }
 
+    uint64_t get_network_handle(const std::string& network_interface) noexcept
+    {
+#if __ANDROID__ && __ANDROID_API__ >= 21
+        using namespace rpp::jni;
+
+        Class Activity{"android/app/Activity"};
```

Comment:
You need to wrap everything around try/catch, because the JNI stuff will throw exceptions and will abort the app.
Better to return some known invalid value and document it.

For example, -1 should be documented as the invalid value I guess? 0 is valid I think

Reply:
The problem is that `getNetworkHandle` returns unsigned 64 bit, where 0 is returned when interface is unavailable, otherwise it can be any other value:
```
 if (netId == 0) {
    return 0L;  // make this zero condition obvious for debugging
}
return (((long) netId) << HANDLE_MAGIC_SIZE) | HANDLE_MAGIC;
```
While on other platforms the handle type is int, and in most cases 0 is default handle, which is valid.

Maybe we can declare these methods as android only for now, since it is not implemented for other platforms?

Reply:
std::optional<uint64_t>